architecture/ambient/ztunnel.md

```diff
@@ -0,0 +1,227 @@
+* Be lightweight enough to not limit adoption.
+  * This puts a much tighter budget on CPU, memory, latency, and throughput requirements than traditional Istio.
+
+Ztunnel was not designed to be a feature-rich data plane.
+Quite the opposite - an *aggressively* small feature set is the key feature that makes ztunnel viable.
+It very intentionally does not offer L7 (HTTP) functionality, for instance, which would likely violate some of the goals above, without contributing to them.
+Instead, the rich functionality that service mesh is traditionally associated with is deferred to the waypoints.
+The ztunnel is primarily a mechanism to get traffic to the waypoints, securely.
+
+## Proxy implementation
+
+In its initial implementations, the ztunnel was actually implemented in 3 different ways: a bespoke Rust implementation, a bespoke Go implementation, and in Envoy.
+
+In the end, [after evaluation](https://docs.google.com/document/d/1c2123cKuYsBDpIon9FFdctWTUIMFweSjgwG7r8l3818/edit), the decision was to move forward with a Rust implementation.
+This offered performance benefits that were too large to leave on the table, as well as opportunities to tune to our specific needs.
+
+## Configuration protocol
+
+Ztunnel, of course, needs to be dynamically configured in order to make decisions on how it should handle traffic.
+For this purpose, we chose to use xDS transport protocol due to our expertise and existing infrastructure, and because the protocol is well suited to our needs.
+
+However, while we chose to use the xDS *transport protocol*, we chose to not use the xDS resource types, such as Clusters and Listeners.
+In our experience and testing, these types force us to represent data in inefficient ways because they are general purpose.
+Ztunnel is not general purpose; it has an extremely tight goal.
+We can exploit this to make a more efficient protocol, which is critical to achieve our resource footprint goals.
+
+For example, configuring Istio mTLS in Envoy takes roughly 50 lines of JSON (it is in Protobuf, of course, but still relevant).
+Because Ztunnel can have Istio semantics baked in, we do not need to encode all this information on the wire.
+Instead, an Istio specific field like `ExpectedTLSIdentity: spiffe://foo.bar` can encode the same information, at a fraction of the cost.
+
+In our testing, even the most generous representations give custom types a 10x edge (in size, allocations, and CPU time) over Envoy types.
+In addition, they are more clear and strictly typed; using Envoy types would require us to put a lot of information in untyped `metadata` maps.
+
+With this in mind, Ztunnel supports two xDS resources: `Address` and `Authorization`.
+
+### Address Type
+
+The primary configuration consumed by Ztunnel is the [`Address` resource](../../pkg/workloadapi/workload.proto).
+As the name suggests, an `Address` represents a particular IP Address.
+This can be a Service or a Workload.
+
+The address type has the following goals:
```

Comment:
We should probably mention that the Address resource will tell you if an address has a waypoint. We refer to this functionality implicitly really far down